src/rdkafka_metadata.c

```diff
@@ -980,7 +1027,8 @@ rd_kafka_parse_Metadata0(rd_kafka_broker_t *rkb,
          * which may contain only a sub-set of the subscribed topics (namely
          * the effective subscription of available topics) as to not
          * propagate non-included topics as non-existent. */
-        if (cgrp_update && (requested_topics || all_topics))
+        if (cgrp_update &&
+            (requested_topics || requested_topic_ids || all_topics))
                 rd_kafka_cgrp_metadata_update_check(rkb->rkb_rk->rk_cgrp,
```

Comment:
Same here. There are multiple places where we are using topic_name in the function or nested functions.. Either we fix those places or add atleast a FIXME here to check where topic name could cause segfault and fix it. Currently, topic id is used in metadata in only new protocol so the code works fine for now.

Reply:
We've checked all the paths and it wasn't causing problems, currently the topic name can be NULL only in parsed `rd_kafka_metadata_topic_internal_t` when the request is made by topic id, we have to make sure it's handled correctly and that it's not inserted in cache or anywhere else in that case, just that it's returned in the response of the Metadata call

Reply:
I agree with this. If we are making sure that this is the case, we are good here. We might need to check more cases later when we implement KIP-516 more.